internal/ignition/templates/discovery.ign

```diff
@@ -0,0 +1,187 @@
+{
+  "ignition": {
+    "version": "3.1.0"{{if .PROXY_SETTINGS}},
+    {{.PROXY_SETTINGS}}{{end}}
+  },
+  "passwd": {
+    "users": [
+      {{.userSshKey}}
+    ]
+  },
+  "systemd": {
+    "units": [{
+      "name": "agent.service",
+      "enabled": {{if .EnableAgentService}}true{{else}}false{{end}},
+      "contents": {{ executeTemplate "agent.service" . | toJson }}
+    },
+    {
+        "name": "selinux.service",
+        "enabled": true,
+        "contents": {{ executeTemplate "selinux.service" . | toJson }}
+    }{{if .OKDBinaries | not}},
+    {
+        "name": "multipathd.service",
+        "enabled": true
+    }{{end}}{{if .StaticNetworkConfig}},
+    {
+        "name": "pre-network-manager-config.service",
+        "enabled": true,
+        "contents": {{ executeTemplate "pre-network-manager-config.service" . | toJson }}
+    }{{end}}{{if .OKDBinaries}},
+    {
+        "name": "okd-overlay.service",
+        "enabled": true,
+        "contents": {{ executeTemplate "okd-overlay.service" . | toJson }}
+    },
+    {
+        "name": "systemd-journal-gatewayd.socket",
+        "enabled": true,
+        "contents": {{ executeTemplate "systemd-journal-gatewayd.socket" . | toJson }}
+    }{{end}}
+    ]
+  },
+  "storage": {
+    "files": [{
+      "overwrite": true,
+      "path": "/usr/local/bin/agent-fix-bz1964591",
+      "mode": 755,
+      "user": {
+          "name": "root"
+      },
+      "contents": { "source": "data:,{{.AGENT_FIX_BZ1964591}}" }
+    },
+    {
+      "overwrite": true,
+      "path": "/etc/motd",
+      "mode": 420,
+      "user": {
+          "name": "root"
+      },
+      "contents": { "source": "data:,{{.AGENT_MOTD}}" }
+    }{{if .OKDBinaries | not}},
+    {
+      "overwrite": true,
+      "path": "/etc/multipath.conf",
+      "mode": 420,
+      "user": {
+          "name": "root"
+      },
+      "contents": { "source": "data:text/plain;charset=utf-8;base64,ZGVmYXVsdHMgewogICAgdXNlcl9mcmllbmRseV9uYW1lcyB5ZXMKICAgIGZpbmRfbXVsdGlwYXRocyB5ZXMKICAgIGVuYWJsZV9mb3JlaWduICJeJCIKfQpibGFja2xpc3RfZXhjZXB0aW9ucyB7CiAgICBwcm9wZXJ0eSAiKFNDU0lfSURFTlRffElEX1dXTikiCn0KYmxhY2tsaXN0IHsKfQo=" }
```

Comment:
Could we get this one out into a file as well? Could be nice

Reply:
Yes, that makes sense. In order to do that I'd suggest to create a new `toDataUrl` template function. That way the `multipath.conf` file could be included like this:

```
"contents": { "source": "{{ executeTemplate "multipath.conf" . | toDataUrl }}" }
```

I did that in a separate pull request (#4052) to not add noise to this one. If you want I can add the relevant [commit](https://github.com/openshift/assisted-service/pull/4052/commits/408b6547a5f6cb38f53e79dc2802500306dbb03e) to this pull request.